src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -483,6 +559,62 @@ internal FunctionPointerMethodSymbol ApplyNullableTransforms(byte defaultTransfo
             }
         }
 
+        public ImmutableHashSet<CustomModifier> GetCallingConventionModifiers()
+        {
+            if (_lazyCallingConventionModifiers is null)
+            {
+                var modifiersToSearch = RefKind != RefKind.None ? RefCustomModifiers : ReturnTypeWithAnnotations.CustomModifiers;
+                if (modifiersToSearch.IsEmpty || CallingConvention != CallingConvention.Unmanaged)
+                {
+                    Interlocked.CompareExchange(ref _lazyCallingConventionModifiers, ImmutableHashSet<CustomModifier>.Empty, null);
+                }
+                else
+                {
+                    var builder = PooledHashSet<CustomModifier>.GetInstance();
+                    foreach (var modifier in modifiersToSearch)
+                    {
+                        if (IsCallingConventionModifier(modifier))
+                        {
+                            builder.Add(modifier);
+                        }
+                    }
+
+                    if (builder.Count == 0)
+                    {
+                        Interlocked.CompareExchange(ref _lazyCallingConventionModifiers, ImmutableHashSet<CustomModifier>.Empty, null);
+                    }
+                    else
+                    {
+                        Interlocked.CompareExchange(ref _lazyCallingConventionModifiers, builder.ToImmutableHashSet(), null);
+                    }
+
+                    builder.Free();
+                }
+            }
+
+            return _lazyCallingConventionModifiers;
+        }
+
+        private static bool IsCallingConventionModifier(CustomModifier modifier)
+        {
+            var modifierType = ((CSharpCustomModifier)modifier).ModifierSymbol;
+            return (object)modifierType.ContainingAssembly == modifierType.ContainingAssembly.CorLibrary
+                   && modifierType.Name.StartsWith("CallConv", ignoreCase: false, culture: CultureInfo.InvariantCulture)
```

Comment:
> modifierType.Name.StartsWith("CallConv", ignoreCase: false, culture: CultureInfo.InvariantCulture) [](start = 21, length = 99)

Do we have a test where a modifier has the right name, but is generic?

Reply:
Added `GenericCallingConventionType`.

---
In reply to: [458923748](https://github.com/dotnet/roslyn/pull/46110#discussion_r458923748) [](ancestors = 458923748)